rust/src/chain_crypto/hash.rs

```diff
@@ -135,9 +137,96 @@ macro_rules! define_hash_object {
                 bech32::to_bech32_from_bytes::<Self>(self.as_ref())
             }
         }
+        impl cbor_event::de::Deserialize for $hash_ty {
```

Comment:
No idea about how to add serializable types. Please review @rooooooooob

Reply:
We actually use the trait in [`utils::Deserialize`](https://github.com/dcSpark/cardano-multiplatform-lib/blob/develop/rust/src/utils.rs#L963) trait to contain additional error information and such, but that trait is automatically implemented by anything that implements `cbor_event::de::Deserialize` so this is fine. It will wrap it in `DeserializeFailure::CBOR` and will contain `None` for location information (i.e. won't tell you in what part of the CBOR it was found) For simple types like a hash this isn't an issue since there's no member name to tag it with anyway, so you're only losing the struct name being tagged but honestly I don't think is a big deal.

For serialization we directly use the `cbor_event` trait so this is exactly what you need to do.

Keep in mind that if you want any of these types to **directly** have [`to_bytes()`/`from_bytes()`](https://github.com/dcSpark/cardano-multiplatform-lib/blob/develop/rust/src/utils.rs#L13) that are was-exposable we need to use those auto-deriving macros in `utils` but this is perfectly fine for the traits. If they're only called from other rust code just the traits is totally fine and will work with the existing (de)serialization system.